design/one-pager-ignore-changes.md

```diff
@@ -0,0 +1,199 @@
+      desiredSize: 1
+      maxSize: 5
+      minSize: 1
+    updateConfig:
+      maxUnavailablePercentage: 50
+      force: true
+  providerConfigRef:
+    name: example
+```
+
+The solution should affect just the Update part of the Crossplane lifecycle, so create and late initialization should work 
+as before, even though a field is marked under `ignoreChanges`. Additionally, as Crossplane will soon support `status.atProvider`,
+users will be able to observe the actual values of the ignored fields.
+
+#### Ignoring required fields
+
+Required fields should not be put into `ignoreChanges`, as that could compromise the functionality of the provider. If such
+a need arises, it should be handled as a special case on the provider side, and not through the `ignoreChanges` feature.
+
+Of course, some fields could be marked as required, but they are actually required only for create. As there currently is
+no standard on differentiating them, it is advised to use `ignoreChanges` on required fields only if certain that it's not
+required for update.
+
+### Implementation
+
+The implementation would be done on the provider level, so we would need to update all the providers over time to use the 
+new field. 
+
+For providers using Upjet the solution will be to leverage Terraform's [ignore_changes] lifecycle field. So Upjet
+generation should be updated with a step that transfers the `spec.ignoreChanges` field into Terraform's `ignore_changes`.
+Something like this in the Upjet method that writes the main TF file:
+
+```go
+// WriteMainTF writes the content main configuration file that has the desired
+// state configuration for Terraform.
+func (fp *FileProducer) WriteMainTF() error {
+
+    fp.parameters["lifecycle"] = map[string]interface{}{
+        // If the resource is in a deletion process, we need to remove the deletion
+        // protection.
+        "prevent_destroy": !meta.WasDeleted(fp.Resource),
+        // Add fields which should be ignored on updates.
+        "ignore_changes": fp.Resource.GetIgnored(),
+    }
+...
+```
+
+For other providers we can implement a helper function in the [crossplane-runtime] repo that would use the existing code
+in [crossplane-runtime/fieldpath] to unset the fields of a managed resource that are under `ignoreChanges`. That way
+provider maintainers could use this function in the Update method without the need to invent their own solution.
+
+Helper function would look something this (WIP):
+```go
+func UnsetIgnored(managed resource.Managed) error {
+	p, err := PaveObject(managed)
+	if err != nil {
+		return err
+	}
+	
+	for _, s := range managed.GetIgnored() {
+		err = p.DeleteField(s)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+```
+
+#### Feature Gating
+
+Similar to all other new features being added to Crossplane, we will ship this
+new policy as an alpha feature that will be off by default and will be
+controlled by the `--enable-alpha-ignore-changes` flag in Providers.
```

Comment:
The CRDs are statically generated and baked into the provider images so we can't really use a flag to add/remove a field. It could only affect whether the code that would process it should be activated or not.

Reply:
Right. My plan was to introduce it in the CRD, but enable/disable it per flag in the providers. The option with annotations that you mentioned [in the comment below](https://github.com/crossplane/crossplane/pull/3822/files#r1127672514) would make sense as well if we want to avoid impact before we are sure about this solution